src/stats/lexd.rs

```diff
@@ -0,0 +1,57 @@
+use std::collections::HashSet;
+
+use tree_sitter::{Language, Parser, TreeCursor};
+
+use rocket_contrib::{json, json::JsonValue};
+
+use slog::Logger;
+
+use crate::{models::StatKind, stats::StatsError};
+
+extern "C" {
+    fn tree_sitter_lexd() -> Language;
+}
+
+pub fn get_stats(_logger: &Logger, body: &str) -> Result<Vec<(StatKind, JsonValue)>, StatsError> {
+    let mut parser = Parser::new();
+    let language = unsafe { tree_sitter_lexd() };
+    parser.set_language(language).unwrap();
```

Comment:
Let's avoid `unwrap` whenever possible, it panics. Instead, could you use `map_err(...)?` here to map the error into a `StatsError` (which is just a wrapper type, you can add to it).